### DISCUSSION TITLE
Autoclear on AutoCompleteBox

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#10908] [Q&A]
How would I go about clearing the text in autocomplete box? 

Currently, let's say I have a list of countries - I type in "Pol", Poland pops in the dropdown, I click it and it updates the selected item to POLAND, however the POL that I typed in the box still remains. I tried setting the SelectedItem to null right after it was updated, basically like this:

```
 public string? SearchedName
 {
       get { return null; }
       set { SetCountry(value); OnPropertyChanged(nameof(SearchedName)); }
  }
```

but that doesn't seem to be doing anything, the letters I input still remain.

Once the country is set I would like the autocomplete to clear and also preferably lose focus.

**Comment by timunie** [comment]
Sounds like you want to create a behavior that handles this. 

1. Listen to Selection changed event 
2. If 
   a. the user hits enter or
   b. the user clicked an item or 
   c. the textbox lost focus, you
3. then update the Text to whatever you want to display there. Most likely it's just `SelectedItem.ToString()`

Custom Behavior sample: https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/master/src/Avalonia.Xaml.Interactions.Custom/ShowOnDoubleTappedBehavior.cs
